### PR TITLE
Upgrade HBase community connector for Dremio 4.7.3

### PR DESCRIPTION
**Pull request by tifflhl**
- Updated README.md to reference Dremio release 4.7.3-202008270723550726-918276ee.
- Updated converters to use org.apache.arrow.memory.ArrowBuf.

### CODE REVIEW

**Review by keeratsingh — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -20,14 +20,14 @@
...
-  <version>4.2.2-202004211133290458-b550b6fa</version>
+  <version>4.7.3-202008270723550726-918276ee</version>
   <artifactId>dremio-hbase-plugin</artifactId>
 
   <name>Dremio HBase Community Connector</name>
 
   <properties>
     <hbase.TestSuite>**/HBaseTestsSuite.class</hbase.TestSuite>
-    <version.dremio>4.2.2-202004211133290458-b550b6fa</version.dremio>
+    <version.dremio>4.7.3-202008270723550726-918276ee</version.dremio>
```
Comment (jacques-n) [inline pom.xml]: let's move this to ${project.version} as opposed to duplicating.

File: README.md [hunk 2]
```diff
@@ -7,9 +7,9 @@
 To add the Dremio Community HBase Connector to the list of Dremio sources:
 
 1. From the project root, run `mvn clean install`. 
-   The HBase Connector `dremio-hbase-plugin-4.2.2-202004211133290458-b550b6fa.jar` appears in the `target` directory.
+   The HBase Connector `dremio-hbase-plugin-4.7.3-202008270723550726-918276ee.jar` appears in the `target` directory.
```
Comment (jacques-n) [inline README.md]: I suggest that we replace with a token such as <version string> so we don't have to update each time we upgrade.

File: pom.xml (hunk 1 above)
Comment (tifflhl) [reply pom.xml]: Done.

File: README.md (hunk 2 above)
Comment (tifflhl) [reply README.md]: I read through a couple Github markdown guides, I didn't find a way to make static text variables. Can you elaborate what you meant be tokenizing the version number?

Comment (jacques-n) [reply README.md]: I just meant put placeholder text and let the user figure out the number.

Comment (tifflhl) [reply README.md]: Updated.

**Pull request merged**